cdk/lib/stack.py

```diff
@@ -22,7 +22,18 @@ def __init__(
 
         super().__init__(scope, id, **kwargs)
 
-        vpc = ec2.Vpc(self, "CtfVPC", max_azs=2)
+        preferred_az = f"{aws.AWS_REGION}a"
+
+        vpc = ec2.Vpc(
+            self,
+            "CtfVPC",
+            max_azs=2,
+            nat_gateways=1,
+            nat_gateway_subnets=ec2.SubnetSelection(
+                availability_zones=[preferred_az],
+                subnet_type=ec2.SubnetType.PUBLIC,
+            ),
```

Comment:
But apparently you can still select only 1 NAT gateway and just force everything to be in the same availability zone so you don't have to pay any cross az data transfer costs